changelog_unreleased/handlebars/pr-6464.md

```diff
@@ -0,0 +1,31 @@
+#### Fix spaces in dynamic element class name ([#6464](https://github.com/prettier/prettier/pull/6464) by [@FabHof](https://github.com/FabHof))
+
+<!-- prettier-ignore -->
+```hbs
+{{!-- Input --}}
+<div class="  hello {{if goodbye true}}{{if goodbye false}} hello{{if goodbye true}}   hello   {{if goodbye false}}hello {{if goodbye true}} hello  ">
+  hello
+</div>
+
+{{!-- Prettier stable --}}
+<div
+  class="hello {{if goodbye true}}{{if goodbye false}} hello{{
+    if goodbye true
+  }} hello {{if goodbye false}}hello {{if goodbye true}} hello"
+>
+  hello
+</div>
+
+{{!-- Prettier master --}}
+<div
+  class="hello
+    {{if goodbye true}}{{if goodbye false}}
```

Comment:
@FabHof you could be interested by this previous discussion: https://github.com/prettier/prettier/pull/7173#discussion_r365378172

For class attributes, it might be okay to insert line breaks but for some other attributes it is not. For instance in (taken from the quoted discussion):
```html
<button title="foo bar">1</button>
<button title="foo
bar">2</button>
```

Reply:
Damn - so is the only option to not format attributes at all? 
Or should there be a whitelist/blacklist for attributes?

Reply:
> is the only option to not format attributes at all?

As far attributes are concerned, I think we can still format things inside mustaches. But texts should be left untouched.

> Or should there be a whitelist/blacklist for attributes?
I'm not really qualified to answer. I would say no because html printer doesn't do that.

Reply:
HTML will format `style` and `class` in 2.0.

Reply:
Nice 👍